### PR TITLE
File Specific TOML for GPTMD

### PR DESCRIPTION
**Pull request by dippman**

### CODE REVIEW

File: TaskLayer/SearchTask/SearchTask.cs [hunk 1]
```diff
@@ -624,6 +624,7 @@ public static void WriteMzidentml(IEnumerable<Psm> items, List<ProteinGroup> gro
             writer.Close();
         }
 
+        //this method will be used by all tasks
```
Comment (stefanks) [inline TaskLayer/SearchTask/SearchTask.cs]: move to MetaMorpheus task! And make protected.

File: TaskLayer/GPTMDTask/GPTMDTask.cs [hunk 2]
```diff
@@ -205,6 +210,19 @@ private IEnumerable<double> GetObservedMasses(IEnumerable<ModificationWithMass>
             }
         }
 
+        private HashSet<DigestionParams> GetListOfDistinctDigestionParams(CommonParameters commonParameters, IEnumerable<CommonParameters> enumerable)
```
Comment (stefanks) [inline TaskLayer/GPTMDTask/GPTMDTask.cs]: move to MetaMorpheus task! And make protected.

File: Test/TestToml.cs [hunk 3]
```diff
@@ -58,11 +58,15 @@ public static void TestTomlForSpecficFiles()
             Assert.AreEqual(InitiatorMethionineBehavior.Undefined, f.InitiatorMethionineBehavior);
             Assert.IsNull(f.MaxMissedCleavages);
 
-            CommonParameters c = SearchTask.SetAllFileSpecificCommonParams(new CommonParameters(), f);
+            //will need to move this/implement runSpecific() in this class
+            /*
+            CommonParameters c = MetaMorpheusTask.SetAllFileSpecificCommonParams(new CommonParameters(), f);
```
Comment (stefanks) [inline Test/TestToml.cs]: Ah, okay! Set it back to public, sorry! And uncomment this test

File: TaskLayer/SearchTask/SearchTask.cs [hunk 4]
```diff
@@ -801,7 +767,7 @@ protected override MyTaskResults RunSpecific(string OutputFolder, List<DbForTask
                         #region Generate indices for modern search
 
                         Status("Getting fragment dictionary...", new List<string> { taskId });
-                        var indexEngine = new IndexingEngine(proteinListSubset, variableModifications, fixedModifications, ionTypes, currentPartition, SearchParameters.SearchDecoy, ListOfDigestionParams, CommonParameters.TotalPartitions, new List<string> { taskId });
+                        var indexEngine = new IndexingEngine(proteinListSubset, variableModifications, fixedModifications, ionTypes, currentPartition, SearchParameters.SearchDecoy, ListOfDigestionParams, combinedParams.TotalPartitions, new List<string> { taskId });
```
Comment (stefanks) [inline TaskLayer/SearchTask/SearchTask.cs]: good catch

Comment (dippman) [reply TaskLayer/SearchTask/SearchTask.cs]: thanks!

**Pull request merged**